droid-results/src/main/java/uk/gov/nationalarchives/droid/profile/datawriter/ColumnBasedDataWriter.java

```diff
@@ -0,0 +1,164 @@
+        int maxIdCount = getMaxIdentificationCount(nodes);
+        for (ProfileResourceNode node : nodes) {
+            List<String> nodeEntries = new ArrayList<>();
+            addNodeColumnsInDefaultOrder(nodeEntries, node);
+            List<Format> formatIdentifications = node.getFormatIdentifications();
+            for (int i = 0; i < maxIdCount; i++) {
+                addColumn(nodeEntries, CsvWriterConstants.HEADER_NAME_PUID,  (i < formatIdentifications.size()) ?  formatIdentifications.get(i).getPuid() : CsvWriterConstants.EMPTY_STRING);
+                addColumn(nodeEntries, CsvWriterConstants.HEADER_NAME_MIME_TYPE, (i < formatIdentifications.size()) ?  formatIdentifications.get(i).getMimeType() : CsvWriterConstants.EMPTY_STRING);
+                addColumn(nodeEntries, CsvWriterConstants.HEADER_NAME_FORMAT_NAME, (i < formatIdentifications.size()) ?  formatIdentifications.get(i).getName() : CsvWriterConstants.EMPTY_STRING);
+                addColumn(nodeEntries, CsvWriterConstants.HEADER_NAME_FORMAT_VERSION, (i < formatIdentifications.size()) ?  formatIdentifications.get(i).getVersion(): CsvWriterConstants.EMPTY_STRING);
+            }
+            csvWriter.writeRow(nodeEntries);
+        }
+        csvWriter.flush();
+    }
+
+    @Override
+    public void writeDataRowsForOneRowPerFormat(List<? extends ProfileResourceNode> nodes, CsvWriter csvWriter) {
+        for (ProfileResourceNode node : nodes) {
+            for (Format format : node.getFormatIdentifications()) {
+                List<String> nodeEntries = new ArrayList<>();
+                addNodeColumnsInDefaultOrder(nodeEntries, node);
+                addColumn(nodeEntries, CsvWriterConstants.HEADER_NAME_PUID, format.getPuid());
+                addColumn(nodeEntries, CsvWriterConstants.HEADER_NAME_MIME_TYPE, format.getMimeType());
+                addColumn(nodeEntries, CsvWriterConstants.HEADER_NAME_FORMAT_NAME, format.getName());
+                addColumn(nodeEntries, CsvWriterConstants.HEADER_NAME_FORMAT_VERSION, format.getVersion());
+                csvWriter.writeRow(nodeEntries);
+            }
+        }
+        csvWriter.flush();
+    }
+
+    @Override
+    public void writeHeadersForOneRowPerFile(List<? extends ProfileResourceNode> nodes, String[] headers, CsvWriter csvWriter) {
+        super.setCustomisedHeaders(headers);
+        List<String> headersToWrite = new ArrayList<>(getHeadersToWrite(getCustomisedHeaders()));
+        int maxIdCount = getMaxIdentificationCount(nodes);
+
+        //if we are writing one row per file, then we tag the "per format" fields as additional columns,
+        //if such columns need to be added, we create appropriate headers with a running suffix and write
+        //them to the file
+        if (!Collections.disjoint(headersToWrite, CsvWriterConstants.PER_FORMAT_HEADERS)) {
+            if (maxIdCount > 1) { //add headers
```

Comment:
Could this condition be added to the if statement above? Should save a few characters.

Reply:
Done